collect/cache/span_cache_basic.go

```diff
@@ -127,6 +130,10 @@ func (sc *SpanCache_basic) GetTraceIDs(n int) []string {
 	if sc.nextix+n > len(sc.current) {
 		n = len(sc.current) - sc.nextix
 	}
+	defer func() {
```

Comment:
Ohhhhhh, this is clever. Maybe too clever? It at least needs an explanatory comment, please, here and in the other place you use it.